pkg/detectors/jdbc/jdbc.go

```diff
@@ -30,38 +33,23 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 
 	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
 	for _, match := range matches {
-		if match[1] == "" {
-			continue
-		}
-		token := match[0]
-		password := match[1]
-
-		//TODO if username and password are the same, username will also be redacted... I think this is  probably correct.
-		redact := strings.TrimSpace(strings.Replace(token, password, strings.Repeat("*", len(password)), -1))
+		jdbcConn := match[0]
 
 		s := detectors.Result{
 			DetectorType: detectorspb.DetectorType_JDBC,
-			Raw:          []byte(token),
-			Redacted:     redact,
+			Raw:          []byte(jdbcConn),
+			Redacted:     tryRedactAnonymousJDBC(jdbcConn),
 		}
 
-		//if verify {
-		//	// TODO: can this be verified? Possibly. Could triage verification to other DBMS strings
-		//	s.Verified = false
-		//	client := common.SaneHttpClient()
-		//	req, err := http.NewRequestWithContext(ctx, "GET", "https://jdbcci.com/api/v2/me", nil)
-		//	if err != nil {
-		//		continue
-		//	}
-		//	req.Header.Add("Accept", "application/json;")
-		//	req.Header.Add("Jdbc-Token", token)
-		//	res, err := client.Do(req)
-		//	if err == nil {
-		//		if res.StatusCode >= 200 && res.StatusCode < 300 {
-		//			s.Verified = true
-		//		}
-		//	}
-		//}
+		if verify {
+			s.Verified = false
+			j, err := newJDBC(jdbcConn)
+			if err != nil {
+				continue
+			}
+			s.Verified = j.ping()
```

Comment:
Don't we need a timeout here?

Reply:
Yeah, the timeout is hidden in each implementation of the `ping()` method, usually by calling this helper function:

```go
func pingErr(driverName, conn string) error {
	db, err := sql.Open(driverName, conn)
	if err != nil {
		return err
	}
	defer db.Close()

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second) // <---------
	defer cancel()
	if err := db.PingContext(ctx); err != nil {
		return err
	}
	return nil
}
```

I'll surface it, good callout.